### ISSUE TITLE
Docs: Not clear how to use "request_attributes"

### ISSUE THREAD

**Issue by tfrokt** [envoyproxy/envoy#32678]
I am struggling reading the docs. It has no .yaml (or JSON) examples which makes it really hard.

I have successfully added an ext_proc [1] to my envoy proxy, but the the `path` does not reach the external processor (it does reach the cluster). I had a look at `request_attributes`, but it is not clear to me if that would solve my issue or even what the correct syntax is.

E.g. is this correct? If yes, what does it do?

```
http_filters:
          - name: envoy.filters.http.ext_proc
            typed_config:
              "@type": "type.googleapis.com/envoy.extensions.filters.http.ext_proc.v3.ExternalProcessor"
              failure_mode_allow: true
              async_mode: false
              message_timeout:
                seconds: 5
                nanos: 0
              processing_mode:
                request_header_mode: "SEND"
                response_header_mode: "SEND"
                request_body_mode: "NONE"
                response_body_mode: "NONE"
                request_trailer_mode: "SKIP"
                response_trailer_mode: "SKIP"
              request_attributes:
                - "request.path"
              grpc_service:
                google_grpc:
                  target_uri: "<external processor>"
                  stat_prefix: "<metric prefix>"
```

[1] https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_proc/v3/ext_proc.proto

**Comment by KBaichoo** [comment]
cc @tyxia @yanavlasov as codeowners

**Comment by tyxia** [comment]
> I have successfully added an ext_proc [1] to my envoy proxy, but the the path does not reach the external processor (it does reach the cluster)

request_attributes  is for forwarding Envoy request attributes to external processor when matched. In other words, your client request should already have path field in to be matched.
 + @jbohanon  who added attributes feature.

To triage this, if you don't use `request_attributes` feature, just configure  request_header_mode to SEND, does your external processor server receive headers?

**Comment by jbohanon** [comment]
To be totally clear, what version of Envoy are you using? The attributes feature merged very recently and will not be available in a released version of Envoy until 1.30. 

Assuming you are using a version which accepts this config, the yaml you posted looks correct to me. The attribute should appear in [this field](https://github.com/envoyproxy/envoy/blob/3e597f5af0e4604e0bd755e420b3ed4209e1a49c/api/envoy/service/ext_proc/v3/external_processor.proto#L121) in a struct under the key `envoy.filters.http.ext_proc`

**Comment by tfrokt** [comment]
We are running an old version (1.26.4), I should have checked .. 

@tyxia I can confirm `request_header_mode: SEND` works.

Thanks for the quick reply all!
